src/core/lib/iomgr/ev_poll_posix.cc

```diff
@@ -395,6 +399,28 @@ static void unref_by(grpc_fd* fd, int n) {
 #endif
   gpr_atm old = gpr_atm_full_fetch_add(&fd->refst, -n);
   if (old == n) {
+    /* We have a cache of pollers and their results. Even though we are done
+     * with this fd, we might get an fd with the same number. To avoid using
+     * results from this fd for a future fd, zero out the results for this fd.
+     */
+    gpr_mu_lock(&g_cvfds.mu);
```

Comment:
We should only do this if we are using the poll-cv engine.  The g_cvfds mutex is only initialized if we are using the poll-cv engine I believe.

Reply:
I just realized that :)